osu.Game/Screens/Play/GameplayClockContainer.cs

```diff
@@ -36,92 +38,101 @@ public class GameplayClockContainer : Container, IAdjustableClock, IGameplayCloc
 
         /// <summary>
         /// The time from which the clock should start. Will be seeked to on calling <see cref="Reset"/>.
+        /// Settting a start time will <see cref="Reset"/> to the new value.
         /// </summary>
         /// <remarks>
-        /// If not set, a value of zero will be used.
-        /// Importantly, the value will be inferred from the current ruleset in <see cref="MasterGameplayClockContainer"/> unless specified.
+        /// By default, a value of zero will be used.
+        /// Importantly, the value will be inferred from the current beatmap in <see cref="MasterGameplayClockContainer"/> by default.
         /// </remarks>
-        public double? StartTime { get; set; }
+        public double StartTime
+        {
+            get => startTime;
+            set
+            {
+                startTime = value;
+                Reset();
```

Comment:
Don't know if I agree with `StartTime`'s setter having a side effect of resetting the clock. I wouldn't expect that to happen if I didn't read the implementation/xmldoc. If there is a need to do so, then I'd see it becoming a `Reset(double startTime)` method or something?

Reply:
That's what it used to be before a previous refactor. I guess we can go back to that. It doesn't make much sense to me to change the start time without the clock also being reset.

Reply:
I've moved it to be part of the `Reset` call, see how it feels now.